src/OrchardCore.Modules/OrchardCore.ContentTypes/Controllers/AdminController.cs

```diff
@@ -143,14 +142,14 @@ public async Task<ActionResult> CreatePOST(CreateTypeViewModel viewModel)
 
             await _notifier.SuccessAsync(H["The \"{0}\" content type has been created.", typeViewModel.DisplayName]);
 
-            return RedirectToAction("AddPartsTo", new { id = typeViewModel.Name });
+            return RedirectToAction(nameof(AddPartsTo), new { id = typeViewModel.Name });
         }
 
         public async Task<ActionResult> Edit(string id)
         {
-            if (!await _authorizationService.AuthorizeAsync(User, Permissions.EditContentTypes))
```

Comment:
Keep it, and handle the custom permission in the authorization handler.

Reply:
Created a new ContentDefinitionAutherizationHandler to validate content types/parts.